common/models/types/index.d.ts

```diff
@@ -131,7 +160,8 @@ declare interface LexicalModel {
    * @param transform A Transform corresponding to a recent input keystroke
    * @param context A depiction of the context to which `transform` is applied.
    * @returns A probability distribution (`Distribution<Suggestion>`) on the resulting `Suggestion` 
-   * space for use in determining the most optimal overall suggestions.
+   * space for use in determining the most optimal overall suggestions.  Each returned `Suggestion`
+   * should replace the entire word / text token, rather than leaving part of it unaltered.
```

Comment:
At the moment, the assumption this allows makes casing suggestions much, much easier than otherwise.

If you refer to #3824, this allows directly casing the `displayAs` and `insert` strings on the returned `Suggestion`s with little-to-no extra fanfare.  `displayAs` would probably still be fine if this change were dropped, but then the `ModelCompositor` would have to include significant extra logic to properly case the `Suggestion`'s result - especially for `'initial'` case:

- Does `transform.deleteLeft` equal the current token's character count before the `Suggestion` is applied?
  - If not...
  - Extend the `Transform` to now delete and re-insert all currently-unaltered characters from the token.
    - Must modify `deleteLeft` and `insert` accordingly.

I guess it is somewhat straightforward to do, but authoring the initial pass (and reviewing, probably) these two PRs seems easier without adding in that extra layer of complexity.  I could be sure to revert this spec change this in a followup.

-----

In case it helps with clarity, a somewhat forced example:

Current context:  `There`
`suggestion.displayAs`: `Therapist`

`suggestion.transform`, without the spec shift above, could technically be:
```
{
  insert: 'apist'
  deleteLeft: 1
}
```

In #3824's current state, case-matching to 'initial' would then yield `'TherApist'` when applied, which is incorrect.

Apologies about the bleed-over; this change belongs more in the other PR, for sure.

Reply:
Now addressed by #3845.